src/pykeen/datasets/generate.py

```diff
@@ -36,26 +45,26 @@ def main(path: str, directory: str, test_ratios, no_validation: bool, validation
     sub_triples_factories = cast(Sequence[TriplesFactory], triples_factory.split(ratios, random_state=seed))
 
     for subset_name, subset_tf in zip(LABELS, sub_triples_factories):
-        output_path = os.path.join(directory, f'{subset_name}.txt')
-        click.echo(f'Outputing {subset_name} to {output_path}')
+        output_path = directory.joinpath(subset_name).with_suffix(".txt")
+        click.echo(f'Outputing {subset_name} to {output_path.as_uri()}')
         np.savetxt(output_path, subset_tf.triples, delimiter='\t', fmt='%s')
 
     metadata = dict(
-        source=os.path.abspath(path),
+        source=path.resolve(),
```

Comment:
will this get coerced into a string properly

Reply:
No. Maybe it would be better to convert to a string here, instead of waiting until JSON serialization.

Reply:
agreed.